rust/core-lib/src/view.rs

```diff
@@ -920,24 +924,36 @@ impl View {
         self.find.first_mut().unwrap().do_find(text, chars, case_sensitive, is_regex);
     }
 
-    pub fn find_next(&mut self, text: &Rope, reverse: bool, wrap: bool) {
-        self.select_next_occurrence(text, reverse, false);
+    pub fn find_next(&mut self, text: &Rope, reverse: bool, wrap: bool, allow_same: bool,
+                     modify_selection: &SelectionModifier) {
+        self.select_next_occurrence(text, reverse, false, allow_same, modify_selection);
         if self.scroll_to.is_none() && wrap {
-            self.select_next_occurrence(text, reverse, true);
+            self.select_next_occurrence(text, reverse, true, allow_same, modify_selection);
         }
     }
 
     /// Select the next occurrence relative to the last cursor. `reverse` determines whether the
     /// next occurrence before (`true`) or after (`false`) the last cursor is selected. `wrapped`
     /// indicates a search for the next occurrence past the end of the file.
-    pub fn select_next_occurrence(&mut self, text: &Rope, reverse: bool, wrapped: bool)
+    pub fn select_next_occurrence(&mut self, text: &Rope, reverse: bool, wrapped: bool,
+                                  allow_same: bool, modify_selection: &SelectionModifier)
     {
         // select occurrence closest to last selection
         let sel = match self.sel_regions().last() {
-            Some(sel) => (sel.min(), sel.max()),
+            Some(sel) => {
+                if allow_same && sel.is_caret() {
+                    let (start, end) = {
+                        let mut word_cursor = WordCursor::new(text, sel.max());
+                        word_cursor.select_word()
+                    };
+                    (end, start)
```

Comment:
Why is this (end, start) and not (start, end)? 😕❔

Reply:
Okay, cool. I'd welcome a short comment; reading the code it definitely looked intentional but I couldn't figure out the logic.